### PR TITLE
adding first iteration of stat_daemon service

### PR DESCRIPTION
**Pull request by askeys**
@mistercrunch.  It works basically like we discussed (about the same as Airflow.)  I will commit a job definition file and company-specific plugins to our internal repo for separate review.

### CODE REVIEW

File: contrib/services/stat_daemon/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Stat Daemon
+----
+
+`stat_daemon` is a utility that passively monitors for changes to metadata and updates it accordingly.  Current supported data types are hive tables, hdfs directories and s3 buckets.  Data is pushed to a MySQL or sqlite table with the following schema:
+```SQL
+CREATE TABLE {table_name} (
+    type    CHAR(32) NOT NULL,
```
Comment (artwr) [inline contrib/services/stat_daemon/README.md]: Does this work in SQLite? https://www.sqlite.org/datatype3.html

Comment (askeys) [inline contrib/services/stat_daemon/README.md]: Yes, seems to.  This is how I tested it initially.

File: contrib/services/stat_daemon/stat_daemon/plugins/s3_default.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python
+import logging
+import time
+from airflow.hooks import S3Hook
+
+def get_rows(args):
+    """
+    Default stats updater for hdfs
```
Comment (artwr) [inline contrib/services/stat_daemon/stat_daemon/plugins/s3_default.py]: for s3

**Comment by askeys** [comment]
@artwr removed the s3 plugin for now.  We can add back later.

File: contrib/services/stat_daemon/stat_daemon/util.py [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+            stat    CHAR(64),
+            val     FLOAT,
+            ts      INT
+        );
+        """.format(table=self.table_name)
+        logging.info("Executing SQL: \n" + sql)
+        db.run(sql)
+
+
+class Reporter:
```
Comment (mistercrunch) [inline contrib/services/stat_daemon/stat_daemon/util.py]: (object)

File: contrib/services/stat_daemon/setup.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    # Include additional files into the package
+    #include_package_data=True,
+    scripts=['stat_daemon/stat_daemon', ],
+    # Details
+    url="",
+    # license="LICENSE.txt",
+    description="A metadata collection process.",
+    # long_description=open("README.txt").read(),
+    # Dependent packages (distributions)
+    install_requires=[
```
Comment (mistercrunch) [inline contrib/services/stat_daemon/setup.py]: airflow?

**Comment by askeys** [comment]
Closing this out for now, will add it to the new repo in a way we decide is appropriate.

**Comment by mistercrunch** [comment]
Let's use this as a use case for the plugin architecture I'll write soon.

**Comment by askeys** [comment]
Sounds good.

**Pull request closed without merging**